### PR TITLE
2374 feat/handle missing raw headers

### PR DESCRIPTION
**Pull request by macintoshpie**
#### Any background context you want to provide?
Previously, when importing csvs:
- empty header fields were just kept empty as the raw column name
- if there were multiple empty header fields, the last one would essentially 'overwrite' the other empty fields.

#### What's this PR do?
When importing CSVs (not spreadsheets) if there are empty headers, SEED will generate headers for them. Then on the mapping page users are given notice of this with the option to delete the file.

#### How should this be manually tested?
Run the migration and refresh browser cache.
Before each of the following instructions, upload `portfolio-manager-sample-missing-headers.csv`, once on the mapping page you should see a modal warning you that it auto-generated some headers.  

Test 1: click "Continue" on the warning modal, you should see the generated headers as `SEED Generated Header #`. Finish mapping. If you go back to the mapping page you should _not_ see this modal warning.

Test 2: click "Delete File" on the warning modal, you should be redirected to the dataset page and the imported file should be deleted.

#### What are the relevant tickets?
#2374

### CODE REVIEW

**Comment by macintoshpie** [comment]
I did not implement this functionality for excel spreadsheets b/c the way the parser works makes it pretty unlikely for there to be an empty header. Basically, the header of a spreadsheet is determined by finding the first row that has no empty cells.

File: seed/lib/mcm/reader.py [hunk 1]
```diff
@@ -405,40 +428,23 @@ class CSVParser(object):
             # rows.next() will return the first row
     """
 
-    def __init__(self, csvfile, *args, **kwargs):
```
Comment (macintoshpie) [inline seed/lib/mcm/reader.py:408]: I found no instances of *args or **kwargs being used for initialization, so I removed them (and their downstream uses)

**Comment by adrian-lara** [comment]
When running through manual testing - Things seem to work as expected. But during one of my tests, I triggered a 500 when hitting the `/start_` endpoint.

The relevant parts of the error message here:
```
Internal Server Error: /api/v3/import_files/5617/start_system_matching_and_geocoding/
Traceback (most recent call last):
  File "./seed/venv367-django-2-2-10-upgrade-20200220/lib/python3.6/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "seed_propertyview_property_id_f8bdf6c2_uniq"
DETAIL:  Key (property_id, cycle_id)=(1189362, 370) already exists
...
  File "./seed/seed/views/v3/import_files.py", line 457, in start_system_matching_and_geocoding
    return task_match_buildings(pk)
  File "./seed/seed/data_importer/tasks.py", line 1250, in match_buildings
    finish_matching.s(file_pk, progress_data.key))
...
 File "./seed/seed/data_importer/match.py", line 139, in match_and_link_incoming_properties_and_taxlots
    merged_linked_property_views = link_views(merged_property_views, PropertyView)
  File "./seed/seed/data_importer/match.py", line 416, in link_views
    _merge_count, _link_count, view_id = match_merge_link(view.id, state_class_name)
  File "./seed/seed/utils/match.py", line 238, in match_merge_link
    link_count = _link_matches(matching_views, org_id, target_view, ViewClass)
  File "./seed/seed/utils/match.py", line 169, in _link_matches
    view.save()

```

I'm having trouble reproducing, and the error message would indicate the error triggered at a point in the code you didn't change. My UI actions leading up to the error were as follows:
- import missing headers CSV from test directory
- test "delete" feature once prompted on mapping page
- import missing headers CSV from test directory
- successfully map records
- revisit mapping page and see that no prompt provided (👍 )
- import a different spreadsheet without missing headers
- after clicking the Save Mappings on the mapping confirmation screen, I get the 500.

Any ideas? I think this might not be an issue with your PR and likely an issue with the state of my DB during the testing, as I've not been able to reproduce. Still poking around a bit, but wanted to see what you think.

**Review by adrian-lara — APPROVED** [review]
Code review looks good! 👍

**Comment by adrian-lara** [comment]
> When running through manual testing - Things seem to work as expected. But during one of my tests, I triggered a 500 when hitting the `/start_system_matching_and_geocoding` endpoint.

Did some more digging, and couldn't figure out what caused that error. I also wasn't able to reproduce it. I'm fine with merging and just seeing if it comes up again in later testing.

**Comment by RDmitchell** [comment]
Is there a reason we are not implementing this also for spreadsheets (it seems it is only implemented for CSV files)? It is definitely possible for a spreadsheet to have a column with data without a column name at the top.  Maybe that is handled elsewhere for spreadsheets?

**Pull request merged**